In [1]:
# import packages
import os
import sys
import pandas as pd
import numpy as np

In [2]:
# prepare data path to read data
RAW_DATA_PATH = os.path.abspath(os.path.join(os.getcwd(),os.path.pardir,'data','raw'))
# prepare package path to get functions commonly used
PACKAGE_PATH = os.path.abspath(os.path.join(os.getcwd(),os.path.pardir,os.path.pardir,
                                            'Common_Functions','EDA'))

In [3]:
PACKAGE_PATH

'D:\\Machine_Learning\\Common_Functions\\EDA'

In [4]:
# add PACKAGE_PATH to path
sys.path.insert(0,PACKAGE_PATH)

In [6]:
# read csv
shopint_df = pd.read_csv(os.path.join(RAW_DATA_PATH,'online_shoppers_intention.csv'))

In [7]:
# have quick glance of data
shopint_df.head()

Administrative  Administrative_Duration  Informational  \
0             0.0                      0.0            0.0   
1             0.0                      0.0            0.0   
2             0.0                     -1.0            0.0   
3             0.0                      0.0            0.0   
4             0.0                      0.0            0.0   

   Informational_Duration  ProductRelated  ProductRelated_Duration  \
0                     0.0             1.0                 0.000000   
1                     0.0             2.0                64.000000   
2                    -1.0             1.0                -1.000000   
3                     0.0             2.0                 2.666667   
4                     0.0            10.0               627.500000   

   BounceRates  ExitRates  PageValues  SpecialDay Month  OperatingSystems  \
0         0.20       0.20         0.0         0.0   Feb                 1   
1         0.00       0.10         0.0         0.0   Feb                 2   
2         0.20       0.20         0.0         0.0   Feb                 4   
3         0.05       0.14         0.0         0.0   Feb                 3   
4         0.02       0.05         0.0         0.0   Feb                 3   

   Browser  Region  TrafficType        VisitorType  Weekend  Revenue  
0        1       1            1  Returning_Visitor    False    False  
1        2       1            2  Returning_Visitor    False    False  
2        1       9            3  Returning_Visitor    False    False  
3        2       2            4  Returning_Visitor    False    False  
4        3       1            4  Returning_Visitor     True    False

In [8]:
# get basic info about data
shopint_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
Administrative             12316 non-null float64
Administrative_Duration    12316 non-null float64
Informational              12316 non-null float64
Informational_Duration     12316 non-null float64
ProductRelated             12316 non-null float64
ProductRelated_Duration    12316 non-null float64
BounceRates                12316 non-null float64
ExitRates                  12316 non-null float64
PageValues                 12330 non-null float64
SpecialDay                 12330 non-null float64
Month                      12330 non-null object
OperatingSystems           12330 non-null int64
Browser                    12330 non-null int64
Region                     12330 non-null int64
TrafficType                12330 non-null int64
VisitorType                12330 non-null object
Weekend                    12330 non-null bool
Revenue                    12330 non-null bool
dtypes:

In [9]:
# find number of nulls in each column
shopint_df.isnull().sum()

Administrative             14
Administrative_Duration    14
Informational              14
Informational_Duration     14
ProductRelated             14
ProductRelated_Duration    14
BounceRates                14
ExitRates                  14
PageValues                  0
SpecialDay                  0
Month                       0
OperatingSystems            0
Browser                     0
Region                      0
TrafficType                 0
VisitorType                 0
Weekend                     0
Revenue                     0
dtype: int64

**Insights**
1. columns 'Administrative', 'Administrative_Duration', 'Informational','Informational_Duration', 'ProductRelated', 'ProductRelated_Duration','BounceRates', 'ExitRates' have null values
2. These columns have equal number of null values, it is most likely that all columns have null values in same rows.

In [10]:
# checking if all null are in same rows
shopint_df[pd.isnull(shopint_df['Administrative'])].isnull().sum()

Administrative             14
Administrative_Duration    14
Informational              14
Informational_Duration     14
ProductRelated             14
ProductRelated_Duration    14
BounceRates                14
ExitRates                  14
PageValues                  0
SpecialDay                  0
Month                       0
OperatingSystems            0
Browser                     0
Region                      0
TrafficType                 0
VisitorType                 0
Weekend                     0
Revenue                     0
dtype: int64

**Insights**
1. The columns  'Administrative', 'Administrative_Duration', 'Informational','Informational_Duration', 'ProductRelated', 'ProductRelated_Duration','BounceRates', 'ExitRates' have null values in same rows

In [11]:
# find % of null vaules
from null_values import find_null_percent

In [12]:
find_null_percent(shopint_df)

index       0
0            Administrative  0.11 %
1   Administrative_Duration  0.11 %
2             Informational  0.11 %
3    Informational_Duration  0.11 %
4            ProductRelated  0.11 %
5   ProductRelated_Duration  0.11 %
6               BounceRates  0.11 %
7                 ExitRates  0.11 %
8                PageValues   0.0 %
9                SpecialDay   0.0 %
10                    Month   0.0 %
11         OperatingSystems   0.0 %
12                  Browser   0.0 %
13                   Region   0.0 %
14              TrafficType   0.0 %
15              VisitorType   0.0 %
16                  Weekend   0.0 %
17                  Revenue   0.0 %

**Insights**
1. The data has only 0.11% of null values. **These rows can be dropped.**